src/core/common/message.cpp

```diff
@@ -133,7 +142,8 @@ ThreadError Message::ResizeMessage(uint16_t aLength)
     {
         if (curBuffer->GetNextBuffer() == NULL)
         {
-            curBuffer->SetNextBuffer(NewBuffer());
+            curBuffer->SetNextBuffer(NewBuffer(mHeader.mContext));
+            VerifyOrExit(curBuffer->GetNextBuffer() != NULL, ;);
```

Comment:
This fixes a bug that I encountered in my testing where ResizeMessage was invoked while no more buffers where available.